File: w10_d1_0_white_boarding/solutions/1_palindrome.rb
# Problem 1 - Palindrome

def is_palindrome(str)
  # simple solution
  str = str.downcase

  # removing spaces and punctuation
  # str = str.downcase.gsub(/(\s+|\W+)/, '');

  i = 0
  while i < str.length
    if str[i] != str[str.length - i - 1]
      return false
    end
    i += 1
  end

  return true
end

File: w10_d1_0_white_boarding/solutions/1_palindrome.js
// Problem 1 - Palindrome

function isPalindrome(str) {
  // simple solution
  str = str.toLowerCase();

  // removing spaces and punctuation
  // str = str.toLowerCase().replace(/(\s+|\W+)/g, '');

  for (var i = 0; i < str.length; i ++) {
    if (str[i] !== str[str.length - i - 1]) {
      return false;
    }
  }
  return true;
}